flask/app/routes.py

```diff
@@ -341,116 +340,99 @@ def bulk_update():
         created_by_id = 1
 
     for i, row in enumerate(dat):
-
-        # family logic
-
-        fam_query = models.Family.query.filter(
+        # Find the family by codename or create it if it doesn't exist
+        family_id = models.Family.query.filter(
             models.Family.family_codename == row.get("family_codename")
-        )
-
-        if not fam_query.value("family_id"):
-
-            fam_objs = models.Family(
+        ).value("family_id")
+        if not family_id:
+            family = models.Family(
                 family_codename=row.get("family_codename"),
                 created_by_id=created_by_id,
                 updated_by_id=updated_by_id,
             )
-            db.session.add(fam_objs)
-
+            db.session.add(family)
             transaction_or_abort(db.session.flush)
+            family_id = family.family_id
 
-        # participant logic
-
-        # query both participant codename and family id else we will get the participant ID for the first member
-        # of the family and no subsequent entities will be updated
-        ptp_query = models.Participant.query.filter(
-            models.Participant.family_id == fam_query.value("family_id"),
-            models.Participant.participant_codename == row.get("participant_codename"),
-        )
-
+        # Fail if we have any invalid values
         enum_error = enum_validate(
             models.Participant, row, editable_dict["participant"]
         )
-
         if enum_error:
             db.session.rollback()
             return f"Error on line {str(i + 1)} - " + enum_error, 400
 
-        ptp_objs = models.Participant(
-            family_id=fam_query.value("family_id"),
-            participant_codename=row.get("participant_codename"),
-            sex=row.get("sex"),
-            notes=row.get("notes"),
-            affected=row.get("affected"),
-            solved=row.get("solved"),
-            participant_type=row.get("participant_type"),
-            month_of_birth=row.get("month_of_birth"),
-            created_by_id=created_by_id,
-            updated_by_id=updated_by_id,
-        )
-
-        db.session.add(ptp_objs)
-        transaction_or_abort(db.session.flush)
-
-        # tissue logic
-
-        tis_query = models.TissueSample.query.filter(
-            models.TissueSample.participant_id == ptp_query.value("participant_id")
-        )
-
```

Comment:
So the check (398) for an existing tissue sample under a participant was causing additional tissue samples to fail?

Reply:
What happened here was that if the participant already had a tissue sample then it would not proceed further to add any data to do with this row

Reply:
Good catch!